ui/lib/core/addon/components/enable-input.hbs

```diff
@@ -0,0 +1,27 @@
+{{!
+  Copyright (c) HashiCorp, Inc.
+  SPDX-License-Identifier: BUSL-1.1
+~}}
+
+{{#if this.enable}}
+  {{yield}}
+{{else}}
+  <div class="flex" ...attributes>
+    <div class="is-flex-grow-1">
+      {{#if @attr}}
+        <ReadonlyFormField @attr={{@attr}} @value="**********" />
```

Comment:
Using `ReadonlyFormField` passes along the label, subtext and any attribute options rendered by `FormField` so "enabling" the is visually seemless
![azure-edit](https://github.com/hashicorp/vault/assets/68122737/d5a38de5-9535-419f-b335-e488407ab22a)